vundo-diff.el

```diff
@@ -0,0 +1,150 @@
+;; available are:
+;;
+;;    m  mark a node for diff
+;;    u  unmark the marked node
+;;    d  diff between the marked (or parent) and current nodes
+;;    
+;; All of the normal navigation command of vundo work as normal.
+
+;;; Code:
+(require 'vundo)
+(require 'diff)
+
+(defface vundo-diff-highlight
+  '((((background light)) .
+     (:inherit vundo-node :foreground "DodgerBlue4"))
+    (((background dark)) .
+     (:inherit vundo-node  :foreground "DodgerBlue1")))
+  "Face for nodes marked for diff in the undo tree.")
+
+(defvar-local vundo-diff--marked-node nil)
+(defvar-local vundo-diff--highlight-overlay nil
+  "Overlay used to highlight the selected node.")
+
+;;;###autoload
+(defun vundo-diff-mark (&optional node)
+  "Mark NODE for vundo diff.
+NODE defaults to the current node."
+  (interactive)
+  (let* ((mod-list vundo--prev-mod-list)
+         (node (or node (vundo--current-node mod-list))))
+    (setq vundo-diff--marked-node node)
+    (unless vundo-diff--highlight-overlay
+      (setq vundo-diff--highlight-overlay
+            (make-overlay (1- (vundo-m-point node)) (vundo-m-point node)))
+      (overlay-put vundo-diff--highlight-overlay
+                   'display (vundo--translate "●"))
+      (overlay-put vundo-diff--highlight-overlay
+                   'face 'vundo-diff-highlight)
+      (overlay-put vundo-diff--highlight-overlay 'priority 2))
+    (move-overlay vundo-diff--highlight-overlay
+                  (1- (vundo-m-point node))
+                  (vundo-m-point node))))
+
+;;;###autoload
+(defun vundo-diff-unmark ()
+  "Unmark the node marked for vundo diff."
+  (interactive)
+  (when vundo-diff--marked-node
+    (setq vundo-diff--marked-node nil)
+    (when vundo-diff--highlight-overlay
+      (delete-overlay vundo-diff--highlight-overlay)
+      (setq vundo-diff--highlight-overlay nil))))
+
+;;;###autoload
+(defun vundo-diff ()
+  "Perform diff between marked and current buffer state.
+Displays in a separate diff buffer with name based on
+the original buffer name."
+  (interactive)
+  (let* ((orig vundo--orig-buffer)
+         (oname (buffer-name orig))
+         (current (vundo--current-node vundo--prev-mod-list))
+         (marked (or vundo-diff--marked-node (vundo-m-parent current)))
+         (swapped (> (vundo-m-idx marked) (vundo-m-idx current)))
+         dbuf)
+    (if (or (not current) (not marked) (eq current marked))
+        (message "vundo diff not available.")
+      (let ((mrkbuf (get-buffer-create
+                     (make-temp-name (concat oname "-vundo-diff-marked")))))
+        (vundo--move-to-node current marked orig vundo--prev-mod-list)
+        (with-current-buffer mrkbuf
+          (insert-buffer-substring-no-properties orig))
+        (vundo--refresh-buffer orig (current-buffer) 'incremental)
+        (vundo--move-to-node marked current orig vundo--prev-mod-list)
+        (vundo--refresh-buffer orig (current-buffer) 'incremental)
```

Comment:
I want to double-check.  Is it your understanding that you need to `(vundo--refresh-buffer ... 'incremental)` after every `move-to-node` like this?   Somehow I got the impression that this is needed only once (i.e. you can move as much as you like).  But I get `aref` and `no route` errors without both of these calls.

Reply:
I take it back.  I get `no route` without a `refresh-buffer` after each move.  Restored that.  Any idea why?